google-cloud-spanner-executor/src/main/java/com/google/cloud/executor/spanner/CloudExecutorImpl.java

```diff
@@ -48,6 +56,33 @@ public StreamObserver<SpannerAsyncActionRequest> executeActionAsync(
       @Override
       public void onNext(SpannerAsyncActionRequest request) {
         LOGGER.log(Level.INFO, String.format("Receiving request: \n%s", request));
+
+        // Use Multiplexed sessions for all supported operations if the
+        // multiplexedSessionOperationsRatio from command line is > 0.0
+        if (multiplexedSessionOperationsRatio > 0.0) {
+          SessionPoolOptions.Builder sessionPoolOptionsBuilder;
+          if (request.getAction().getSpannerOptions().hasSessionPoolOptions()) {
```

Comment:
Add a check like `request.getAction().hasSpannerOptions()` before doing `getSpannerOptions`. Else this will throw a NullPointerException.

Reply:
The autogenerated methods from the proto always returns a default instance of the class and not `null`. In this case `getAction()`, `getSpannerOptions` etc will never be `null`. Hence, we are safe from getting an `NullPointerException `
Reference: https://github.com/googleapis/java-spanner/blob/6f522dd73123b3de8399902e28d755a5496829c3/proto-google-cloud-spanner-executor-v1/src/main/java/com/google/spanner/executor/v1/SpannerAsyncActionRequest.java#L112-L115

Reply:
Ok, got it.